File: JavaPOP/src/pages/MainPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class MainPage {

    private WebDriver driver;

    @FindBy(id = "lst-ib")
    private WebElement searchField;
    @FindBy(name = "btnK")
    private WebElement searchButton;
    @FindBy(xpath = "//a[text() ='QADev']")
    private WebElement linkForPage;

    //constructor
    public MainPage(WebDriver driver)
    {
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }

    //method
    public void findQAdevPage(){
        searchField.sendKeys("qadev.pl");
        searchButton.click();
    }

    public void chooseQAdevPage(){
        linkForPage.click();
    }

    public void close()
    {
        this.driver.close();
    }
}

File: C#POP/SeleniumTest/SeleniumTest/PageObject/MainPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SeleniumTest.PageObject
{
    class MainPage
    {
        private IWebDriver driver;

        [FindsBy(How = How.Id, Using = "lst-ib")]
        [CacheLookup]
        public IWebElement searchField { get; set; }

        [FindsBy(How = How.Name, Using = "btnK")]
        [CacheLookup]
        public IWebElement searchButton { get; set; }

        [FindsBy(How = How.XPath, Using = "//a[text() ='QADev']")]
        [CacheLookup]
        public IWebElement linkForPage { get; set; }
        
        public MainPage(IWebDriver driver)
        {
            this.driver = driver;
            PageFactory.InitElements(driver, this);
        }

        //method
        public void findQAdevPage()
        {
            searchField.SendKeys("qadev.pl");
            searchButton.Click();
        }

        public void chooseQAdevPage()
        {
            linkForPage.Click();
        }

        public void close()
        {
            this.driver.Close();
        }
    }
}
